azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/queryer/MavenPluginQueryerDefaultImpl.java

```diff
@@ -0,0 +1,123 @@
+package com.microsoft.azure.maven.queryer;
+
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugin.logging.Log;
+import org.codehaus.plexus.util.StringUtils;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.util.List;
+
+public class MavenPluginQueryerDefaultImpl extends MavenPluginQueryer {
+
+    public static final String CLOSE_FAILURE_WARNING = "Can't close io stream for queryer.";
+    public static final String FOUND_VALID_VALUE = "Found valid value. Skip user input.";
+    public static final String PROMPT_STRING_WITH_DEFAULTVALUE = "Define value for %s(Default: %s): ";
+    public static final String PROMPT_STRING_WITHOUT_DEFAULTVALUE = "Define value for %s: ";
+    public static final String DEFAULT_INPUT_ERROR_MESSAGE = "Invalid input, please check and try again.";
+    public static final String DEFAULT_INPUT_ERROR_MESSAGE_WITH_REGEX = "Invalid input, value should match( %s ).";
+
+    private BufferedReader reader;
+    private PrintWriter writer;
+    private Log log;
+
+    public MavenPluginQueryerDefaultImpl(Log log) {
+        this(System.in, System.out, log);
+    }
+
+    public MavenPluginQueryerDefaultImpl(InputStream inputStream, OutputStream outputStream, Log log) {
+        this.log = log;
```

Comment:
The constructor creates 4 readers and writers but in the `close()` method, only two of them are get closed